CMakeLists.txt

```diff
@@ -133,6 +134,10 @@ if(CMAKE_BUILD_TYPE MATCHES Release)
   endif()
 endif()
 
+if (COVERAGE)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ftest-coverage")
```

Comment:
I'm going to add a check for `CMAKE_BUILD_TYPE == Debug`. Because that is the only sensible combination.